src/releaseNotesProvider.ts

```diff
@@ -0,0 +1,243 @@
+  return makeRight(undefined);
+
+  // below are helper functions
+
+  async function showPanel(version: string, releaseNotesUrl: string) {
+    const panel = vscode.window.createWebviewPanel(
+      `scalameta.metals.whatsNew`,
+      `Metals ${version} release notes`,
+      vscode.ViewColumn.One
+    );
+
+    const releaseNotes = await getReleaseNotesMarkdown(
+      releaseNotesUrl,
+      context,
+      (uri) => panel.webview.asWebviewUri(uri)
+    );
+
+    panel.webview.html = releaseNotes;
+    panel.reveal();
+
+    // Update current device's latest server version when there's no value or it was a older one.
+    // Then sync this value across other devices.
+    state.update(versionKey, version);
+    state.setKeysForSync([versionKey]);
+
+    context.subscriptions.push(panel);
+  }
+
+  /**
+   * Don't show panel for remote environment because it installs extension on every time.
+   * TODO: what about wsl?
+   */
+  function isRemote(): Either<string, void> {
+    // const isWsl = env.remoteName === "wsl";
+    return env.remoteName == null
+      ? makeRight(undefined)
+      : makeLeft(`is a remote environment ${env.remoteName}`);
+  }
+
+  /**
+   *  Return version for which release notes should be displayed
+   */
+  function getVersion(): Either<string, string> {
+    const previousVersion: string | undefined = state.get(versionKey);
+    // strip version to
+    // in theory semver.clean can return null, but we're almost sure that currentVersion is well defined
+    const cleanVersion = semver.clean(currentVersion);
+
+    if (cleanVersion == null) {
+      const msg = `can't transform ${currentVersion} to 'major.minor.patch'`;
+      return makeLeft(msg);
+    }
+
+    if (!previousVersion) {
+      // if there was no previous version then show release notes for current cleaned version
+      return makeRight(currentVersion);
+    }
+
+    const compare = semver.compare(cleanVersion, previousVersion);
+    const diff = semver.diff(cleanVersion, previousVersion);
+
+    // take into account only major, minor and patch, ignore snapshot releases
+    const isNewerVersion =
+      compare === 1 &&
+      (diff === "major" || diff === "minor" || diff === "patch");
+
+    return isNewerVersion
+      ? makeRight(cleanVersion)
+      : makeLeft("do not show release notes for an older version");
+  }
+}
+
+/**
+ * Translate server version to link to the markdown file with release notes.
+ * @param version clean version in major.minor.patch form
+ * If version has release notes return link to them, if not return nothing.
+ * Sample link to which we're doing request https://api.github.com/repos/scalameta/metals/releases/tags/v0.11.6.
+ * From such JSON obtain body property which contains link to the blogpost, but what's more important,
+ * contains can be converted to name of markdown file with release notes.
+ */
+async function getMarkdownLink(
+  version: string
+): Promise<Either<string, string>> {
```

Comment:
This function might be seems magic but as long as Github Release contains link to blogpost this will work and there is no need to specify release notes manually.